Kj/all

Katinka666 opened the pull request.

Katinka666 committed:
everything

Katinka666 committed:
checkstyle changes

Katinka666 committed:
handle json converter exception

rbocsi commented on .checkstyle:
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
In the future projects this file should be on the .gitignore list (we've done that in the AH5)

rbocsi commented on .classpath:
```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
In the future projects this file should be on the .gitignore list (we've done that in the AH5)

rbocsi commented on .project:
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
In the future projects this file should be on the .gitignore list (we've done that in the AH5)

rbocsi commented on README.md:
```diff
@@ -1 +1,31 @@
\ No newline at end of file
+# Arrowhead Authorization Ruler
+
+Updates the authorization rules within a local cloud based on systems' metadata and service definitions using the SysOp certificate of a local cloud.
```
systems' metadata =>systems' name or metadata

rbocsi commented on README.md:
````diff
@@ -1 +1,31 @@
+```
+[
+   {
+      "source":"<sys-metadata-key>=<sys-metadata-value>",
````
we renamed source to consumer

rbocsi commented on README.md:
```diff
@@ -1 +1,31 @@
+[
+   {
+      "source":"<sys-metadata-key>=<sys-metadata-value>",
+      "target":"<sys-metadata-key>=<sys-metadata-value>",
```
we renamed source to consumer

rbocsi commented on pom.xml:
```diff
@@ -0,0 +1,81 @@
+	
+	<url>https://github.com/Aitia-IIOT/ah-auth-ruler-script</url>
+	<licenses>
+    	<license>
+	     	<name>Eclipse Public License - v 2.0</name>
+	     	<url>https://www.eclipse.org/org/documents/epl-2.0/EPL-2.0.txt</url>
+	   </license>
```
@borditamas

Not sure about this license.  I don't think the Eclipse Foundation has any business with this program.

borditamas replied:
Agree. Let's use MIT:

```
<license>
      <name>MIT License</name>
      <url>http://www.opensource.org/licenses/mit-license.php</url>
    </license>
```

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,397 @@
+	@Autowired
+	private SSLProperties sslProperties;
+
+	@Value("#{${default_interface_name_list}}")
```
the expression here should be a constant.
Also, I think ${default_interface_name_list} should be enough to load a list from properties

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,397 @@
+			SpringApplication.run(ConsumerMain.class, args);
+		}
+	}
+
```
missing assistant methods header

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,397 @@
+		updateAuthRules(newRules);
+	}
+
+	// -------------------------------------------------------------------------------------------------
```
lots of missing finals (Ctrl+A, Ctrl+1, Enter)

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,397 @@
+		for (int id : ruleIdsToDelete) {
+			logger.info("Removing authorization rule with id: " + id);
+			String response = arrowheadService.consumeServiceHTTP(String.class, HttpMethod.DELETE,
+					Utilities.createURI(authorizationUri.get("scheme"), authorizationUri.get("host"),
```
all these keys should be constants

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,397 @@
+	}
+
+	// -------------------------------------------------------------------------------------------------
+	private List<AuthRule> getRules(final String filename) {
```
assistant methods header should be here

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,397 @@
+
+		for (int i = 0; i < rules.size(); i++) {
+
+			providerId = new ArrayList<>();
+			serviceDefinitionId = new ArrayList<>();
```
these lists are always contains only one element. You should set the initial capacity to 1 in the constructor.

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,397 @@
+		}
+
+		for (int id : ruleIdsToDelete) {
+			logger.info("Removing authorization rule with id: " + id);
```
should be debug instead of info

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,397 @@
+	// ------------------------------------------------------------------------------------------------
+	public static void main(final String[] args) {
+		if (args.length != 1) {
+			System.out.println("You must specify exactly one path! (E.g.: ../example.json)");
```
all System.out.println should be logger.info or logger.error

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,397 @@
+							Integer.parseInt(authorizationUri.get("port")), authorizationUri.get("path")
+									+ ConsumerConstants.QUERY_AUTH_INTRA_CLOUD + "/" + Integer.toString(id)),
+					null, null);
+			logger.info("Http DELETE response: " + response);
```
should be debug instead of info

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,397 @@
+		if (response == null) {
+			logger.error("Could not apply the following authorization rule: " + ruleToAdd.toString());
+		} else {
+			logger.info("Successfully applied rule with id: " + response.getData().getFirst().getId());
```
should be debug instead of info

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,397 @@
+			logger.error(e.getMessage());
+		}
+
+		logger.info("Sending the POST request for the following authorization rule: " + ruleToAdd.toString());
```
should be debug instead of info

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,397 @@
+	// the system string can be systemname or metadata
+	// (in case it is metadata, it must contain '=')
+	private long getSystemIdByInfo(final String systemInfo) throws Exception {
+		Assert.notNull(this.systems, "The ist of possible systems is empty!");
```
typo here: ist => list (?)

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,397 @@
+	private long getSystemIdByInfo(final String systemInfo) throws Exception {
+		Assert.notNull(this.systems, "The ist of possible systems is empty!");
+		for (SystemResponseDTO system : systems) {
+			if (systemInfo.contains("=")) {
```
= should be a constant

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,397 @@
+		Assert.notNull(this.systems, "The ist of possible systems is empty!");
+		for (SystemResponseDTO system : systems) {
+			if (systemInfo.contains("=")) {
+				String[] metaData = systemInfo.split("=");
```
metaData should be metadata (just for consistency with arrowhead)

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,397 @@
+					throw new IllegalArgumentException("System metadata is in invalid format!");
+				}
+				String key = metaData[0];
+				String vaule = metaData[1];
```
typo vaule => value

also you should trim() the key and value

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,397 @@
+	// -------------------------------------------------------------------------------------------------
+	// the system string can be systemname or metadata
+	// (in case it is metadata, it must contain '=')
+	private long getSystemIdByInfo(final String systemInfo) throws Exception {
```
This method is wrong. In case of metadata, it only returns the first system that matches that metadata. It should return all.

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,397 @@
+	private long interfaceNameToId(final String interfaceName) throws Exception {
+		Assert.notNull(this.interfaces, "The list of possible interfaces is empty!");
+		for (ServiceInterfaceResponseDTO interfaceElement : this.interfaces) {
+			if (interfaceElement.getInterfaceName().equals(interfaceName)) {
```
should trim() interfaceName

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,397 @@
+						&& system.getMetadata().get(key).equals(vaule)) {
+					return system.getId();
+				}
+			} else if (system.getSystemName().equals(systemInfo)) {
```
should trim() systemInfo

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,397 @@
+
+	// -------------------------------------------------------------------------------------------------
+	private long serviceDefinitionToId(final String serviceDefinition) throws Exception {
+		Assert.notNull(this.services, "The ist of possible services is empty!");
```
typo: ist => list

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,397 @@
+	private long serviceDefinitionToId(final String serviceDefinition) throws Exception {
+		Assert.notNull(this.services, "The ist of possible services is empty!");
+		for (ServiceDefinitionResponseDTO service : this.services) {
+			if (service.getServiceDefinition().equals(serviceDefinition)) {
```
should trim() serviceDefinition

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,397 @@
+	}
+
+	// -------------------------------------------------------------------------------------------------
+	private List<Integer> getRuleIdsToDelete(final List<AuthorizationIntraCloudResponseDTO> rules, final List<Integer> systemIds) {
```
in case of ids you should use Long/long everywhere

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,397 @@
+
+	// -------------------------------------------------------------------------------------------------
+	private List<SystemResponseDTO> getSystems() {
+		logger.info("Get systems request started...");
```
info => debug

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,397 @@
+		SystemListResponseDTO response = arrowheadService.consumeServiceHTTP(SystemListResponseDTO.class,
+				HttpMethod.GET,
+				Utilities.createURI(getScheme(),
+						CommonConstants.$SERVICEREGISTRY_ADDRESS_WD.split(":")[1].replace("}", ""),
```
This is really strange and only works if your Service Registry is on the localhost. What you need is the value of the `@Value(CommonConstants.$SERVICEREGISTRY_ADDRESS_WD)`. (This should be a member)

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,397 @@
+				HttpMethod.GET,
+				Utilities.createURI(getScheme(),
+						CommonConstants.$SERVICEREGISTRY_ADDRESS_WD.split(":")[1].replace("}", ""),
+						Integer.parseInt(CommonConstants.$SERVICEREGISTRY_PORT_WD.split(":")[1].replace("}", "")),
```
similar situation than in the previous comment.

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,397 @@
+		ServiceDefinitionsListResponseDTO response = arrowheadService.consumeServiceHTTP(
+				ServiceDefinitionsListResponseDTO.class, HttpMethod.GET,
+				Utilities.createURI(getScheme(),
+						CommonConstants.$SERVICEREGISTRY_ADDRESS_WD.split(":")[1].replace("}", ""),
+						Integer.parseInt(CommonConstants.$SERVICEREGISTRY_PORT_WD.split(":")[1].replace("}", "")),
```
same here

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,397 @@
+		ServiceInterfacesListResponseDTO response = arrowheadService.consumeServiceHTTP(
+				ServiceInterfacesListResponseDTO.class, HttpMethod.GET,
+				Utilities.createURI(getScheme(),
+						CommonConstants.$SERVICEREGISTRY_ADDRESS_WD.split(":")[1].replace("}", ""),
+						Integer.parseInt(CommonConstants.$SERVICEREGISTRY_PORT_WD.split(":")[1].replace("}", "")),
```
and here

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,397 @@
+
+		if (response == null) {
+			logger.error("Getting the authorization rules was unsuccessful!");
+		} else if (response.getData().isEmpty()) {
```
I am not sure it is a problem if there are no current authorization rules. It just means you don't have to delete anything before you add the new ones.

borditamas replied:
It's not a problem, but it's worth to have a debug log about this case

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/data/AuthRule.java:
```diff
@@ -0,0 +1,79 @@
+	//=================================================================================================
+	// methods
+
+	public AuthRule() {
```
missing //------------------------------------------------------------------------------------------------- (need one before every method)

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/data/AuthRule.java:
```diff
@@ -0,0 +1,79 @@
+	public AuthRule() {
+	}
+
+	//=================================================================================================
```
use single line instead of double

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/data/AuthRule.java:
```diff
@@ -0,0 +1,79 @@
+		this.interfaces = interfaces;
+	}
+
+	//=================================================================================================
```
use single line instead of double

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/data/AuthRule.java:
```diff
@@ -0,0 +1,79 @@
+		return this.interfaces;
+	}
+
+	//=================================================================================================
```
use single line instead of double

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/data/AuthRule.java:
```diff
@@ -0,0 +1,79 @@
+		this.interfaces = interfaces;
+	}
+
+	//=================================================================================================
+	public void printOut() {
+		System.out.println("consumer: " + consumer);
+		System.out.println("provider: " + provider);
+		System.out.println("service: " + service);
+		System.out.println("interfaces: ");
+		System.out.print("\t");
+		for (String i : interfaces) {
+			System.out.print(i + " ");
+		}
+		System.out.println();
+	}
```
we don't need this method. You should use the toString() if you want to print it. Also, do not use System.out.println(), use logger instead.

rbocsi commented on src/main/resources/application.properties:
```diff
@@ -0,0 +1,38 @@
+###           DEFAULT VALUES             ###
+############################################
+
+default_interface_name_list= {'HTTP-SECURE-JSON','HTTP-INSECURE-JSON'}
```
if this is a simple comma separated list, you don't need the difficult SpEL expression in the main class.

rbocsi reviewed (changes requested):
See inline comments

Katinka666 committed:
fix changes according to review

- fixing wrong logic: multiple consumer and provider can match to metadata string, not only one!
- other small changes (e.g. constants, formatting, logging)

Katinka666 committed:
update licence in pom.xml

Katinka666 committed:
Merge branch 'development' into kj/all

borditamas commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,435 @@
+	private void deleteRules(final List<AuthRule> rules) {
+		List<Long> ruleIdsToDelete = new ArrayList<Long>();
+		try {
+			ruleIdsToDelete = getRuleIdsToDelete(getAuthorizationRules(), getSystemIdsToDelete(rules));
```
`getAuthorizationRules()` can retun null value and `getRuleIdsToDelete()` will throw NullPointerException which has the error message "null" what is not really informative

borditamas commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,435 @@
+		} else {
+			rules = response.getData();
+		}
+		return rules;
```
`rules` could be null, but it is not handled in the caller moethods. Initializing `rules` with `new ArrayList<>()` would be better.

borditamas commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,435 @@
+				}
+				rules.add(newRule);
+			} catch (final Exception e) {
+				logger.error("Could not create authorization rule (invalid json format): " + e.getMessage());
```
Throw an exception or call a terminate method, otherwise you just logging, but the cycle goes forward.
Or set `rules=null` and call `break`, since you handling null value in the caller method.

Katinka666 replied:
If the json format is incorrect (for example braces or commas are missing), I'm throwing an exception and returning `null` above, as you suggested.
Otherwise, I am creating the `AuthRule `objects one by one. In some cases It migth not violate the json format, but cannot be interpreted as an `AuthRule `object, for example:
```
[
   {
      "consumeeeer":"testconsumer1",
      "provider":"systemtype=test",
      "service":"test-service",
      "interfaces":[
      ]
   },
   {
      "consumer":"testconsumer2",
      "provider":"testprovider",
      "service":"test-service",
      "interfaces":[
      ]
   }
]
```
It this case, there is a typo in the first rule, so it logs out that the first rule cannot be created, but the cycle goes forward and creates the second one.
My intention was that if one rule was in an invalid format, still apply the others. Should I modify the code so no rules will be applied if one of them is incorrect?

borditamas replied:
> My intention was that if one rule was in an invalid format, still apply the others. Should I modify the code so no rules will be applied if one of them is incorrect?

I think yes you should. This is a user input, it can be expected to being completely adequate.

borditamas commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,435 @@
+						authorizationUri.get(ConsumerConstants.PATH) + ConsumerConstants.QUERY_AUTH_INTRA_CLOUD),
+				null, null);
+
+		if (response == null) {
```
Probably `response` never will be null (`arrowheadService.consumeServiceHTTP()` throws an excpetion if no response), but if I'm not right,  logging is not enough either. You should terminate if existing rules cannot be fetched.

borditamas commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,435 @@
+	private void addSingleRule(final AuthorizationIntraCloudRequestDTO ruleToAdd) {
+		Map<String, String> authorizationUri = null;
+		try {
+			authorizationUri = getAuthorizationUri();
```
Authorization URI stays the same, once you have retrieved it. No need for always go through on all the systems to find it. Should be cached  

Plus, if you haven't found the uri, you shoud terminate.

borditamas commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,435 @@
+		}
+
+		logger.debug("Sending the POST request for the following authorization rule: " + ruleToAdd.toString());
+		final AuthorizationIntraCloudListResponseDTO response = arrowheadService.consumeServiceHTTP(
```
`arrowheadService.consumeServiceHTTP()` throws exception, and if it does not just the single rule won't be saved, but the whole execution will terminate because it is not handled.

Here you shoud put it in try-catch block and check the error. If the error is 500 or 401, than just throw forward and let the execution terminate. But if it 400 you should just log the error message. (Maybe a related system was unregistered meanwhile or something similar)

borditamas commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,435 @@
+					result.add(system.getId());
+				}
+			}
+			//case: sisteminfo is systemname
```
typo

borditamas commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerConstants.java:
```diff
@@ -0,0 +1,30 @@
+	public static final String QUERY_GET_SYSTEMS = "/mgmt/systems";
+	public static final String QUERY_GET_SERVICES = "/mgmt/services";
+	public static final String QUERY_GET_INTERFACES = "/mgmt/interfaces";
+	public static final String QUERY_AUTH_INTRA_CLOUD = "/mgmt/intracloud";
```
`QUERY_AUTH_INTRA_CLOUD` constant name is misleading
`/mgmt/intracloud` is not just for query, you also use it for create rules with POST method.

borditamas commented on src/main/java/eu/arrowhead/application/skeleton/consumer/data/AuthRule.java:
```diff
@@ -0,0 +1,87 @@
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	public void printOut() {
```
No need for this method

Katinka666 committed:
fix mistakes from review

mainly exception handling

Katinka666 committed:
Merge branch 'kj/all' of https://github.com/Aitia-IIOT/ah-auth-ruler-script into kj/all

borditamas commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,452 @@
+			services = getServices();
+			interfaces = getInterfaces();
+			authorizationUri = getAuthorizationUri();
+		} catch (Exception e) {
```
missing final

borditamas commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,452 @@
+			logger.error(e.getMessage());
+			return null;
+		}
+		List<AuthRule> rules = new ArrayList<>();
```
missing final

borditamas commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,452 @@
+		if (deleteRules(newRules)) {
+			try {
+				addRules(newRules);
+			} catch (Exception e) {
```
missing final

borditamas commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -0,0 +1,452 @@
+							Integer.parseInt(authorizationUri.get(ConsumerConstants.PORT)),
+							authorizationUri.get(ConsumerConstants.PATH) + ConsumerConstants.OP_AUTH_INTRA_CLOUD),
+					null, ruleToAdd);
+		} catch (ArrowheadException ae) {
```
missing final

borditamas reviewed (approved):
approved, but see some missing final and the user input related comments

Katinka666 committed:
fix missing finals

Katinka666 committed:
fix exception handling (user input related)

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerConstants.java:
```diff
@@ -1,16 +1,27 @@
 	public static final String QUERY_GET_INTERFACES = "/mgmt/interfaces";
-	public static final String QUERY_AUTH_INTRA_CLOUD = "/mgmt/intracloud";
+	public static final String OP_AUTH_INTRA_CLOUD = "/mgmt/intracloud";
+	public static final List<String> DEFAULT_INTERFACE_NAMES = (List<String>) Arrays.asList(INTERFACE_SECURE, INTERFACE_INSECURE);
```
You can use List.of() to avoid the cast

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -61,138 +62,161 @@ public class ConsumerMain implements ApplicationRunner {
-		} catch (Exception e) {
-			logger.error(e.getMessage());
+		} catch (final Exception e) {
+			logger.error("Finding the authorization rules to delete was unsuccessfull. Reason: " + e.getMessage());
```
typo: unsuccessfull => unsuccessful

rbocsi commented on src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java:
```diff
@@ -61,138 +62,161 @@ public class ConsumerMain implements ApplicationRunner {
+		final List<AuthorizationIntraCloudRequestDTO> dtoList = new ArrayList<>();
+
+		List<Long> consumerIdList; //multiple consumerid can match one auth rule's systeminfo
+		List<Long> prodiverIdList; //multiple providerid can match one auth rule's systeminfo
```
typo prodiver => provider

rbocsi reviewed (approved):
approved, but i have found some typos.

Katinka666 committed:
fix typos

Katinka666 committed:
fix more typos
